core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestOverrideConfiguration.java

```diff
@@ -73,6 +79,43 @@ protected RequestOverrideConfiguration(Builder<?> builder) {
         return apiNames;
     }
 
+    /**
+     * The amount of time to wait for the http request to complete before giving up and timing out. A positive value
+     * enables this feature.
+     *
+     * <p>The request timeout feature doesn't have strict guarantees on how quickly a request is aborted when the timeout is
+     * breached. The typical case aborts the request within a few milliseconds but there may occasionally be requests that
+     * don't get aborted until several seconds after the timer has been breached. Because of this, the request timeout
+     * feature should not be used when absolute precision is needed.
+     *
+     * @see Builder#apiCallAttemptTimeout(Duration)
```

Comment:
Can we have a section here that differentiates between this and `apiCallTimeout`? I see there's one down there already.